src/NinjaTools.FluentMockServer.API/Downstream/RequestMapper.cs

```diff
@@ -0,0 +1,167 @@
+            {
+                return new ErrorResponse<HttpRequestMessage>(new RequestMappingError(ex));
+            }
+        }
+
+        private async Task<HttpContent> MapContentAsync([NotNull] HttpRequest request)
+        {
+            if (request.Body == null || (request.Body.CanSeek && request.Body.Length <= 0))
+            {
+                return null;
+            }
+
+            if (request.Body != null)
+            {
+                request.EnableBuffering();
+            }
+   
+            
+            // Never change this to StreamContent again, I forgot it doesnt work in #464.
+            var content = new ByteArrayContent(await ToByteArray(request.Body));
+            content.Headers.TryAddWithoutValidation("Content-Type", new[] {request.ContentType});
+
+            // if (!string.IsNullOrEmpty(request.ContentType))
+            //     content.Headers.TryAddWithoutValidation("Content-Type", new[] { request.ContentType });
+
+            AddHeaderIfExistsOnRequest("Content-Language", content, request);
+            AddHeaderIfExistsOnRequest("Content-Location", content, request);
+            AddHeaderIfExistsOnRequest("Content-Range", content, request);
+            AddHeaderIfExistsOnRequest("Content-MD5", content, request);
+            AddHeaderIfExistsOnRequest("Content-Disposition", content, request);
+            AddHeaderIfExistsOnRequest("Content-Encoding", content, request);
+
+            return content;
+        }
+
+        private static void AddHeaderIfExistsOnRequest([NotNull] string key, HttpContent content, [NotNull] HttpRequest request)
+        {
+            if (request.Headers != null && request.Headers.ContainsKey(key))
+                content.Headers.TryAddWithoutValidation(key, request.Headers[key].ToList());
+        }
+
+        [NotNull]
+        private static System.Net.Http.HttpMethod MapMethod([NotNull] HttpRequest request)
+        {
+            return new System.Net.Http.HttpMethod(request.Method ?? "GET");
+        }
+
+        [NotNull]
+        private static Uri MapUri(HttpRequest request)
+        {
+            return new Uri(request.GetEncodedUrl());
+        }
+
+        private void MapHeaders([NotNull] HttpRequest request, HttpRequestMessage requestMessage)
+        {
+            foreach (var header in request.Headers)
+            {
+                if (IsSupportedHeader(header))
+                {
+                    requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                }
+            }
+        }
+
+        private bool IsSupportedHeader(KeyValuePair<string, StringValues> header)
+        {
+            return !UnsupportedHeaders.Contains(header.Key.ToLower());
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ReSharper", "All")]
+        private async Task<byte[]> ToByteArray(Stream stream)
+        {
+            using var reader = new StreamReader(stream, Encoding.UTF8, true, 1024, true);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'reader'.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)